### DISCUSSION TITLE
var, let, const 선언이 호이스팅 되는 방식과 이들이 스코프 규칙에 어떤 영향을 미치는지 설명해 주실 수 있나요?

### DISCUSSION THREAD

**Discussion by lledellebell** [diving-javascript/you-dont-know-js-yet#23] [실행 컨텍스트(Execution Context)]
- #18

**Comment by wogus216** [comment]
먼저 호이스팅이란 변수선언이 해당 스코프의 최상단으로 끌어올려진것 처럼 동작하는 JavaScript의 행동

1.var로 선언된 변수는 초기화와 함께 호이스팅됩니다. 이는 변수가 선언된 위치와 관계없이 함수의 시작부분에 메모리에 할당된 다는 것을 의미합니다.
그러나 실제로 할당은 코드에 도달했을 때 이루어짐. 따라서 변수를 선언하기 전에 접근하면 undefined가 반환됨
2. let 과 const 로 선언된 변수는 호이스팅되지만 초기화는 되지 않습니다. 이를 일시적 사각지대(TDZ)라고 하며, 변수 선언 전에 해당 변수를 참조하려고 하면 참조에러가 발생

스코프 규칙

var은 함수 스코프를 가집니다. 즉, var 변수는 선언된 함수 내에서 어디서든지 접근할 수 있으며,함수 밖에서는 접근 할 수 없음.
let과 const는 블록 스코프를 가집니다. 블록 외부에서는 접근할 수 없습니다.
